### PR TITLE
single direction lstm support

### PR DESCRIPTION
**Pull request by pengwa**

### CODE REVIEW

**Comment by pengwa** [comment]
@guschmue which runtime should be used by default in the test?

**Comment by guschmue** [comment]
I think onnxruntime has the most complete onnx support. I switched all defaults to that.
Getting to the pr in the morning.

**Comment by guschmue** [comment]
yes, the onnxruntime currently is compiled with avx and the travis machines don't support avx so I had to disable the backend test in travis. caffe2 also did not use to work and crashes on ub14.04. Waiting for a newer build of onnxruntime .

**Comment by pengwa** [comment]
I see, will ignore the case as well. :)

**Comment by guschmue** [comment]
sorry, forgot to actually post the review.

File: tests/run_pretrained_models.py [hunk 1]
```diff
@@ -167,9 +167,9 @@ def run_tensorflow(self, sess, inputs):
         return result
 
     @staticmethod
-    def to_onnx(tf_graph, opset=None, shape_override=None):
+    def to_onnx(tf_graph, opset=None, shape_override=None, enable_lstm=None):
         """Convert graph to tensorflow."""
-        return process_tf_graph(tf_graph, continue_on_error=False, opset=opset, shape_override=shape_override)
+        return process_tf_graph(tf_graph, continue_on_error=False, opset=opset, shape_override=shape_override, enable_lstm=enable_lstm)
```
[the remark this answers is not in the document]
Comment (pengwa) [reply tests/run_pretrained_models.py]: make sense.

[the remark this answers is not in the document]
Comment (pengwa) [reply tests/run_pretrained_models.py]: the only problem is that I am doing const_fold again here. so using this flag would help enable that. while let me investigate whether const_fold is that necessary.

File: tests/test_lstm.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+        tf.reset_default_graph()
+        g = tf.import_graph_def(graph_def, name='')
+        with tf.Session(graph=g) as sess:
+            g = process_tf_graph(sess.graph, continue_on_error=True, enable_lstm=True) # shape_override={"output:0": [1, 6,4,5]}
+            actual = self._run_backend(g, output_names_with_port, feed_dict)
+
+        for i in range(len(expected)):
+            self.assertAllClose(expected[i], actual[i], rtol=rtol, atol=0.)
+
+    #@unittest.skip("reason for skipping")
```
[the remark this answers is not in the document]
Comment (pengwa) [reply tests/test_lstm.py]: sure

File: tests/test_lstm.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    def test_test_single_dynamic_lstm_stateistuple(self):
+        self.internel_test_single_dynamic_lstm(True)
+
+    #@unittest.skip("reason for skipping")
```
[the remark this answers is not in the document]
Comment (pengwa) [reply tests/test_lstm.py]: removed.

File: tf2onnx/convert.py [hunk 4]
```diff
@@ -34,6 +34,7 @@ def get_args():
     parser.add_argument("--unknown-dim", type=int, default=-1, help="default for unknown dimensions")
     parser.add_argument("--target", default=",".join(DEFAULT_TARGET), help="target platform")
     parser.add_argument("--optimize_transpose", help="eliminate transposes that can be removed", action="store_true")
+    parser.add_argument("--enable_lstm", help="enable lstm support", action="store_true")
```
[the remark this answers is not in the document]
Comment (pengwa) [reply tf2onnx/convert.py]: removed.

File: tests/run_pretrained_models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pengwa) [reply tests/run_pretrained_models.py]: removed that. :)

**Comment by pengwa** [comment]
thanks @guschmue !

**Pull request merged**